include/SFML/Window/Keyboard.hpp

```diff
@@ -153,9 +153,16 @@ enum Key
     F15,          //!< The F15 key
     Pause,        //!< The Pause key
 
-    KeyCount, //!< Keep last -- the total number of keyboard keys
+    Last = Pause, //!< Keep equal to last enumerator -- used to define KeyCount
```

Comment:
What's the point of moving out KeyCount when we still have it in there under a different, imho more confusing name?

If we keep it, we should:
- Not have a comment explaining what it was previously (otherwise just use the old name)
- Have a more speaking name, so nobody confuses it with an actual key name. Like one could easily think `Last` to be some key. So I'd probably with something like `LastKeyEntry` or so, which again begs the question why we can't stick with `KeyCount` to begin with

Reply:
Previously `KeyCount` was listed as any other key, which all have a distinct underlying value. So `sf::Keybaord::KeyCount` was a valid value for the type `sf::Keyboard::Key`, but it did not represent any actual key, which is weird.

Here `Last` is not the same as `KeyCount`. It is defined equal to another valid key value `Pause` so it is not adding an additional bogus value to the enumeration.
The reason I introduced this `Last` enumerator is to make it obvious that it needs to be updated whenever we add more key values by being visually close to where we would add such values. We could as well not have this `Last` enumerator but we would need to be careful to update `KeyCount` definition below whenever we add more values at the end of the enumeration.

What do you prefer? I will be happy to submit another pull request to improve this.

Reply:
I'd rather not duplicate the information. Either store the value in the enum or outside of it. If there was a way to hide a value from the public API, then that might have been the even better option.